### PR TITLE
Improve the install instructions by recommending virtual environments everywhere

### PR DESCRIPTION
**Pull request by MJKirk**

### CODE REVIEW

**Comment by MJKirk** [comment]
Some self-review comments
1) This `\cite` 
https://github.com/eos/eos/blob/fdfa086ee0e37f1cb84331c318c4b3e183867882/doc/installation.rst?plain=1#L194
and `\package`
https://github.com/eos/eos/blob/fdfa086ee0e37f1cb84331c318c4b3e183867882/doc/installation.rst?plain=1#L242
are presumably left over from some old version? Should they be removed or altered?
2) I've written to specifically install ``dynesty==2.0.3``  - I lost track, is this version still specifically required or did we remove the code that broke with newer versions?

File: doc/installation.rst [hunk 1]
```diff
@@ -202,18 +205,17 @@ by running the following commands:
...
 
-::
 
-  # for the 'pip3' command
-  apt-get install python3-pip
-  pip3 install --user dynesty pypmc wilson
-  pip3 install --user --upgrade matplotlib
+::
+  # Make sure to activate your virtual environment first
+  pip install dynesty===2.0.3 matplotlib networkx scipy pypmc pyyaml wilson
```
Comment (mreboud) [inline doc/installation.rst:218]: Three equal signs?

File: doc/installation.rst [hunk 2]
```diff
@@ -247,7 +251,7 @@ To install the remaining packages, run the following command in a shell
 
 ::
 
-  pip3 install matplotlib scipy PyYAML
+  pip3 install dynesty===2.0.3 matplotlib networkx scipy pypmc pyyaml wilson
```
Comment (mreboud) [inline doc/installation.rst:254]: ditto

**Review by mreboud — CHANGES_REQUESTED** [review]

File: doc/installation.rst (hunk 1 above)
Comment (MJKirk) [reply doc/installation.rst:218]: Good catch

Comment (mreboud) [reply doc/installation.rst:218]: I am not 100% sure if this is correct or not. I think `===` has a specific meaning. To be checked :)

Comment (MJKirk) [reply doc/installation.rst]: It does have some specific meaning (see https://packaging.python.org/en/latest/specifications/version-specifiers/#version-matching) but it's not what we want

**Comment by dvandyk** [comment]
> Some self-review comments
> 
> 1. This `\cite`
>    https://github.com/eos/eos/blob/fdfa086ee0e37f1cb84331c318c4b3e183867882/doc/installation.rst?plain=1#L194
>    
>    and `\package`
>    https://github.com/eos/eos/blob/fdfa086ee0e37f1cb84331c318c4b3e183867882/doc/installation.rst?plain=1#L242
>    
>    are presumably left over from some old version? Should they be removed or altered?
Yes, please remove those old LaTeX macros.

> 2. I've written to specifically install `dynesty==2.0.3`  - I lost track, is this version still specifically required or did we remove the code that broke with newer versions?

I think we removed that issue, which was only present in a test case.

**Comment by dvandyk** [comment]
> @dvandyk - what about my comments?

I am certain that I had commented on this. Either Github or the Marseille network seem to have eaten my comments.
 1. Please remove the old LaTeX macros. 
 2. Please remove the version restriction for ``dynesty`` altogether.

**Pull request merged**